src/db.c

```diff
@@ -1806,6 +1833,43 @@ db_query_fetch_group(struct query_params *qp, struct db_group_info *dbgri)
 }
 
 int
+db_query_fetch_count(struct query_params *qp, struct count_info *ci)
+{
+  int ret;
+
+  memset(ci, 0, sizeof(struct count_info));
+
+  if (!qp->stmt)
+    {
+      DPRINTF(E_LOG, L_DB, "Query not started!\n");
+      return -1;
+    }
+
+  if (qp->type != Q_COUNT_ITEMS)
+    {
+      DPRINTF(E_LOG, L_DB, "Not a count query!\n");
+      return -1;
+    }
+
+  ret = db_blocking_step(qp->stmt);
+  if (ret == SQLITE_DONE)
+    {
+      DPRINTF(E_LOG, L_DB, "End of query results for count query\n");
+      return -1;
```

Comment:
Since this is E_LOG and a return of -1 I guess it is an error? I suggest you change the message to reflect that.